apisix/init.lua

```diff
@@ -630,10 +630,11 @@ end
 function _M.http_header_filter_phase()
     core.response.set_header("Server", ver_header)
 
-    local status_code = tonumber(get_var("upstream_status"))
-    if status_code and status_code >= 500 and status_code <= 599 then
-        core.response.set_header("X-APISIX-Upstream-Status", status_code)
-        core.log.info("X-APISIX-Upstream-Status: ", status_code)
+    local up_status = get_var("upstream_status")
+    local from, _ = re_find(up_status, "5[0-9]{2}", "jo")
```

Comment:
We should only care about the last status, not any one happened in retry.

Reply:
Agree, I will update later.

Reply:
@spacewander 
I want to know if the result of `$upstream_status` will appear `502, 200, 502` ?

Reply:
@Firstsawyou 
Normally Nginx won't retry after 200.

Reply:
done.

Reply:
we should use the last status, current way is wrong

Reply:
Yes, I have updated it. Help take a look.